src/Codeception/Step/ConditionalAssertion.php

```diff
@@ -1,26 +1,33 @@
 <?php
+
+declare(strict_types=1);
+
 namespace Codeception\Step;
 
 use Codeception\Exception\ConditionalAssertionFailed;
 use Codeception\Lib\ModuleContainer;
 use Codeception\Util\Template;
+use PHPUnit\Framework\AssertionFailedError;
+use function preg_replace;
+use function str_replace;
+use function strpos;
+use function ucfirst;
 
 class ConditionalAssertion extends Assertion implements GeneratedStep
 {
-    public function run(ModuleContainer $container = null)
+    public function run(ModuleContainer $container = null): void
     {
         try {
             parent::run($container);
-        } catch (\PHPUnit\Framework\AssertionFailedError $e) {
+        } catch (AssertionFailedError $e) {
             throw new ConditionalAssertionFailed($e->getMessage(), $e->getCode(), $e);
         }
     }
 
-    public function getAction()
+    public function getAction(): ?string
```

Comment:
I think that probability of preg_replace failing is so low or even non-existent that the return type of this method should be `: string`

Reply:
It is not acceptable for this preg_replace to fail, so if you think that it can return null, please check if result is null and throw exception.